cli/script/command-executor.ts

```diff
@@ -499,30 +363,31 @@ function deserializeConnectionInfo(): IStandardLoginConnectionInfo | IAccessKeyL
     }
 }
 
-function notifyAlreadyLoggedIn(): Promise<void> {
-    return Q.fcall(() => { throw new Error("You are already logged in from this machine."); });
-}
-
 export function execute(command: cli.ICommand): Promise<void> {
-    connectionInfo = deserializeConnectionInfo();
+    var connectionInfo: ILegacyLoginConnectionInfo|ILoginConnectionInfo = deserializeConnectionInfo();
 
-    switch (command.type) {
-        case cli.CommandType.login:
-            if (connectionInfo) {
-                return notifyAlreadyLoggedIn();
-            }
+    return Q(<void>null)
+        .then(() => {
+            switch (command.type) {
+                case cli.CommandType.login:
+                case cli.CommandType.register:
+                    if (connectionInfo) {
+                        throw new Error("You are already logged in from this machine.");
+                    }
+                    break;
 
-            return login(<cli.ILoginCommand>command);
+                default:
+                    if (!!sdk) break; // Used by unit tests to skip authentication
 
-        case cli.CommandType.logout:
-            return logout(<cli.ILogoutCommand>command);
+                    if (!connectionInfo) {
+                        throw new Error("You are not currently logged in. Run the 'code-push login' command to authenticate with the CodePush server.");
+                    }
 
-        case cli.CommandType.register:
-            return register(<cli.IRegisterCommand>command);
-    }
+                    var accessKey: string = getAccessKeyFromConnectionInfo(connectionInfo);
+                    sdk = new AccountManager(accessKey, userAgent, connectionInfo.serverUrl);
```

Comment:
Need to address the fact that people with older connectionInfo files sitting on disk may still authenticate to the old acquisition server instead of the new management one, even with an upgraded CLI.